### PR TITLE
[Fix](Nereids) fix leading with brace can not generate correct plan

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
Problem:
when using leading like:
leading(t1 {t2 t3} {t4 t5} t6)
it would not generate correct plan because levellist can not express enough message of braces
Solved:
remove levellist express of leading levels and use reverse polish expression
Algorithm:
leading(t1 {t2 t3} {t4 t5} t6)
==>
stack top to down(t1 t2 t3 join join t4 t5 join t6 join)
when generate leading join, we can pop items in stack, when it's a table, make logicalscan when it's a join
operator, make logical join and push back to stack

### CODE REVIEW

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java [hunk 1]
```diff
@@ -87,81 +87,83 @@ public LeadingHint(String hintName) {
...
-                    ConnectContext.get().getStatementContext().addHint(distributeHint);
-                }
-            } else if (parameter.equals("broadcast")) {
-                DistributeHint distributeHint = new DistributeHint(DistributeType.BROADCAST_RIGHT);
-                distributeHints.put(tablelist.size(), distributeHint);
-                if (!ConnectContext.get().getStatementContext().getHints().contains(distributeHint)) {
-                    ConnectContext.get().getStatementContext().addHint(distributeHint);
-                }
+        addJoinParameters = insertJoinIntoParameters(parameters);
+        normalizedParameters = parseIntoPolandNotation(addJoinParameters);
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java:91]: Reverse Polish Notation

**Comment by LiBinfeng-01** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/hint/LeadingHint.java]: done

**Review by morrySnow — APPROVED** [review]

**Review by englefly — APPROVED** [review]

**Pull request merged**